splash/network_manager.py

```diff
@@ -177,6 +178,13 @@ def _handle_custom_proxies(self, request):
         if hasattr(request, 'custom_proxy'):
             self.setProxy(request.custom_proxy)
 
+            user, password = request.custom_proxy.user(), request.custom_proxy.password()
+            if not user and password:
```

Comment:
hmm are you sure it is allowed to have blank user and non-blank password? how proxy can determine password validity without user string? Given bit of confusion around this perhaps safest option is just base64 encoding "user:passowrd" and then setting header if result of b64 is nonempty string.